server/src/main/java/org/opensearch/action/admin/indices/settings/put/TransportUpdateSettingsAction.java

```diff
@@ -106,8 +123,31 @@ protected ClusterBlockException checkBlock(UpdateSettingsRequest request, Cluste
             || IndexMetadata.INDEX_BLOCKS_READ_ONLY_ALLOW_DELETE_SETTING.exists(request.settings())) {
             return null;
         }
-        return state.blocks()
-            .indicesBlockedException(ClusterBlockLevel.METADATA_WRITE, indexNameExpressionResolver.concreteIndexNames(state, request));
+
+        final Index[] requestIndices = indexNameExpressionResolver.concreteIndices(state, request);
```

Comment:
Should we wrap this logic behind the feature flag and get rid of the flag check closer to launch?
I know this code is for launch/GA, but since we already have the overhead 🙂

Reply:
Do you think the check for index fs type is not sufficient?

Reply:
It should be. But I was thinking of preventing block execution all together if the flag is not set or removed before launch.

Reply:
let me make that change